pkg/manager/leaderelection.go

```diff
@@ -57,11 +57,11 @@ func (k *KubeMacPoolManager) UpdateLeaderLabel() error {
 				return errors.Wrap(err, "failed to get kubemacpool manager pod")
 			}
 
-			_, exist := pod.Labels[names.LEADER_LABEL]
+			_, exist := pod.GetLabels()[names.LEADER_LABEL]
 			if pod.Name == k.podName && !exist {
```

Comment:
Adding label or delete it at already added/delete does not fail so we can simplify this by 
```golang
                     if pod.Name == k.podName {
				logger.V(1).Info("add the label to the elected leader", "Pod Name", pod.Name)
				pod.Labels[names.LEADER_LABEL] = "true"
			} else {
				logger.V(1).Info("deleting leader label from old leader", "Pod Name", pod.Name)
				delete(pod.Labels, names.LEADER_LABEL)
			}
```
We can before that is not problem with it at unit-test

Reply:
the logic is simpler but we will update all pods even if there was no label change in them, shouldn't we avoid that?

Reply:
Yaml is going to be the same so they don't get updated also we call `Update` out of the condition.

Reply:
ack